cmake/BLTMacros.cmake

```diff
@@ -1363,6 +1363,134 @@ macro(blt_print_target_properties)
 endmacro(blt_print_target_properties)
 
 
+##------------------------------------------------------------------------------
+## blt_print_variables([NAME_REGEX  <regular_expression_string>]
+##                     [VALUE_REGEX <regular_expression_string>]
+##                     [IGNORE_CASE])
+##
+## Prints all (or filtered) variables and their values in the current scope.
+## 
+## Defaults:
+## The regexes are case-sensitive unless the IGNORE_CASE option is supplied
+## NAME_REGEX = .* (print every property name)
+## VALUE_REGEX = .* (print every property value)
+## 
+##------------------------------------------------------------------------------
+macro(blt_print_variables)
+
+    set(options            IGNORE_CASE)
```

Comment:
`blt` tends to prefer `singleValuedArgs` to `options` in its macros, but `IGNORE_CASE` felt more natural as an `option` (which is either present or absent) than as a `singleValuedArg` which would require `TRUE|FALSE`.

Happy to change it if others feel strongly...